python/python/lance/arrow.py

```diff
@@ -44,15 +44,20 @@
 
 
 class ImageURIType(pa.ExtensionType):
-    def __init__(self):
-        pa.ExtensionType.__init__(self, pa.string(), "lance.arrow.image_uri")
+    def __init__(self, storage_type=pa.string()):
+        # TODO: allow string_view once available?
```

Comment:
Should we just make an `is_string_like` function?  I guess this is only used in one place but might help avoid this situation in the future?

Reply:
Good idea. Added some private functions for this.